// r2t-twitter/lib/r2t-twitter.js

exports.definition = {
    "name": "Twitter",
    "category": "Spouts",
    "subcategory" : "Twitter",
    "storm_type" : "spout",
    "emitedFields" : ["text", "place"],
    "requiredFields" : [],
    "container": {
        "xtype": "WireIt.FormContainer",
        "title": "twitter",
        "fields": [
            {
                "type": "string",
                "name": "consumer_key",
                "wirable": false,
                label: "Consumer Key"
            },
            {
                "type": "string",
                "name": "consumer_secret",
                "wirable": false,
                label: "Consumer Secret"
            },
            {
                "type": "string",
                "name": "access_token",
                "wirable": false,
                label: "Access Token"
            },
            {
                "type": "string",
                "name": "access_token_secret",
                "wirable": false,
                label: "Access Token Secret"
            },
            {
                "type": "list",
                "name": "keywords",
                label: "Keyword(s)",
                elementType: {
                    type: 'string',
                    fields: [],
                    separators: [false, ":", false]
                }
            }
        ],
        "terminals": [
            {
                "name": "out",
                "direction": [0, 1],
                "ddConfig": {
                    "type": "output",
                    "allowedTypes": ["input"]
                }
            }
        ]
    }
};


exports.run = function(params, cb) {
    cb({});
};
